src/renderer.ts

```diff
@@ -204,10 +213,24 @@ class Renderer extends EventEmitter<RendererEvents> {
     return [div, shadow]
   }
 
+  /** Wavesurfer itself calls this method. Do not call it manually. */
   setOptions(options: WaveSurferOptions) {
+    if (this.options.container !== options.container) {
+      const newParent = this.parentFromOptionsContainer(options.container)
+      this.parent.removeChild(this.container)
+      newParent.appendChild(this.container)
+
+      this.parent = newParent
+      this.emit('containerRemount')
+    }
+
+    const renderOptionsChanged = !objectEquals(this.options, options, ['container'], true)
```

Comment:
This could be simplified to something like this:
```js
if (!options.container || Object.keys(options).length > 1) {
  // re-render
}
```

Reply:
Originally wanted to add the `objectEquals` method so that it could be later used for eliminating the need for rerendering when options not related to rendering change such as `dragToSeek`, `autoplay`, `interact` (as far as i know those wont affect the rendering of the waveform themselves but i could be mistaken). If you don't plan on adding those then i'll just remove the utility functions.

Just now realised that there might be an issue with when the `fillParent` option is set to true and the `container` is updated to an element which dimensions does not match the original container's dimensions. This is not problematic for my use case right now but eventually it is very likely that atleast someone else finds it problematic.

Reply:
True, as the only required param, I didn't consider container as a settable option, so some places might break.